daemon: Add info dictionary to cached results

petervo opened the pull request:
I'd like to propose returning a info dictionary from the methods that return cached rpm diffs. This commit right now just changes the xml to the proposed api. If you like the idea then I can fill in the code.

petervo commented:
An alternative would be to replace the UpgradeOrigin property with an UpgradeDetails variant that would have the same data. The nice thing about that is that clients can listen for changes on that property. Of course that doesn't help much with information about rebases. We could of course do both. Any preference?

mbarnes commented:
I like the `UpgradeDetails` idea, but either option is a D-Bus API break.  Maybe that's not a problem though since it doesn't look like the daemon has been officially released yet.

@cgwalters, are we still free to keep tweaking this?

cgwalters commented:
I'd say feel free to keep changing the DBus API until we choose a point to declare it stable.

petervo commented:
@mbarnes I added both

I changed the property name update to CachedUpdate as to would match the similarly named HasCachedUpdate and GetCachedUpdateRpmDiff but we haven't really been consistent everywhere.

In order to get the same info for rebases i added the details variant to the results for GetCachedUpdateRpmDiff and GetCachedRebaseRpmDiff. The update one is a little redundant but i preferred to keep the results of those two methods the same. Feel free to drop the second commit if you don't like this.

cgwalters commented:
This looks reasonable to me.  Can you add a rationale to the commit mesages?  It could be as simple as: "This will be used by the Cockpit interface."

The github linkage to additional information in this conversation is a bit weaker than I'd like because the PR number is part of the merge commit only.

petervo commented:
Updated commit messages.

cgwalters commented:
LGTM, thanks!
